src/makeCalculator.js

```diff
@@ -37,7 +37,50 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {};
+  const errorMessage = 'Error: Division by zero.';
+
+  calculator.result = 0;
+
+  calculator.operate
+    = function(callback, n) {
+      callback(n);
+
+      return calculator;
+    };
+
+  calculator.add
+    = function(n) {
+      calculator.result += n;
+    };
+
+  calculator.subtract
+    = function(n) {
+      calculator.result -= n;
```

Comment:
better to use in `this` function, it will improve the reusability

Reply:
fix it in all places